langkit/sentiment.py

```diff
@@ -1,46 +1,155 @@
 from copy import deepcopy
-from typing import Optional
+from typing import Optional, Set
 
 from whylogs.experimental.core.udf_schema import register_dataset_udf
 from . import LangKitConfig, lang_config, prompt_column, response_column
+from langkit.whylogs.unreg import unregister_udfs
 
 
-_prompt = prompt_column
-_response = response_column
-_sentiment_analyzer = None
-_nltk_downloaded = False
-
-
-def sentiment_nltk(text: str) -> float:
-    if _sentiment_analyzer is None:
+def sentiment_nltk(text: str, sentiment_analyzer) -> float:
```

Comment:
I think we should continue supporting the use of the modules as standalone functions.  It looks like this change would prevent the user from simply calling `sentiment_nltk("I am very sad")`, right?

I think this comment applies to other modules. At least for toxicity, I think it's the same case.

Can we require only the text values for the standalone functions?

Reply:
I don't see a way to have a single function support multiple languages without the extra parameter. There's a few ways it could be curried, but logically it needs to be there.

Reply:
I think having an extra parameter is ok, but requiring the user to pass a SentimentIntensityAnalyzer, for example, looks very complex. Couldn't we have the user be able to pass an optional `language` parameter that defaults to english?

we can choose to drop the single function support, but I personally find it very useful.